common/network-yarn/src/main/java/org/apache/spark/network/yarn/YarnShuffleService.java

```diff
@@ -378,6 +408,18 @@ protected File initRecoveryDb(String dbName) {
         }
       }
     }
+
+    // Find a local_dir which is writable, to avoid creating ldb in a read-only disk.
+    if (_recoveryPath == null) {
+      for (String dir : localDirs) {
+        File f = new File(dir);
+        if (checkFileAvailable(f)) {
+          _recoveryPath = new Path(dir);
+          break;
+        }
+      }
+    }
+
     if (_recoveryPath == null) {
```

Comment:
If `_recoveryPath` is still null I think we should throw an exception here, since none of the disk is good.